ansible/roles/pyxis/tasks/main.yml

```diff
@@ -0,0 +1,30 @@
+---
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: Clone pyxis
+  ansible.builtin.git:
+    repo: 'https://github.com/NVIDIA/pyxis.git'
+    dest: /tmp/pyxis
+    version: 'v{{ pyxis_ver }}'
+
+- name: Build and install pyxis
+  ansible.builtin.shell:
+    cmd: make install
+    chdir: /tmp/pyxis
+
+- name: Add line to plugstack conf
+  ansible.builtin.copy:
+    content: 'required /usr/local/lib/slurm/spank_pyxis.so'
```

Comment:
I think this is replacing the entire content of the file with that, which will remove anything that already exists. Use either:

- https://docs.ansible.com/ansible/latest/collections/ansible/builtin/lineinfile_module.html
- investigate plugstack.conf.d/pyxis.conf (my preference)

Reply:
Yes, the reason I did that was because your script that I was basing this off used a `>` instead of `>>` to append so I assumed you wanted it overwritten.  I will look into this.

Reply:
Done with the lineinfile.  Not sure what you meant by investigation, but I use the regex line `'\s*required\s+/usr/local/lib/slurm/spank_pyxis.so\s*'` to search if it exists and then that line is replaced, or it's added on.  The `create: True` option will create the file if it doesn't exist.